### PR TITLE
Add FIX protocol specification

### PR DESCRIPTION
**Pull request by jvirtanen**
FIX is an alternative order entry protocol for Parity.

### CODE REVIEW

File: parity-fix/doc/FIX.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+
+### Enter Order
+
+Enter an order with a New Order Single message.
+
+Tag               | Req'd | Description
+------------------|-------|-----------------------------------
+_Message Header_  |   Y   | MsgType(35) = D (New Order Single)
+ClOrdID(11)       |   Y   | Unique identifier for the order
```
Comment (penberg) [inline parity-fix/doc/FIX.md]: You should explicitly specify what are the requirements for a valid ``ClOrdID``. For example, if there's a maximum length or limitations on which ASCII characters are allowed.

File: parity-fix/doc/FIX.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+Symbol(55)        |   Y   |
+Side(54)          |   Y   | See below
+TransactTime(60)  |   Y   |
+OrderQty(38)      |   Y   |
```
Comment (penberg) [inline parity-fix/doc/FIX.md]: No ``Price(44)``?

File: parity-fix/doc/FIX.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+OrdType(40)       |   Y   | See below
```
Comment (penberg) [inline parity-fix/doc/FIX.md]: ``Account(1)`` is pretty useful.

File: parity-fix/doc/FIX.md (hunk 2 above)
Comment (jvirtanen) [inline parity-fix/doc/FIX.md]: Well spotted. :+1:

**Comment by jvirtanen** [comment]
Administrative messages, such as Logon, Logout, and Heartbeat are still missing.

File: parity-fix/doc/FIX.md (hunk 1 above)
Comment (jvirtanen) [inline parity-fix/doc/FIX.md]: Good point. :+1:

File: parity-fix/doc/FIX.md (hunk 2 above)
Comment (jvirtanen) [inline parity-fix/doc/FIX.md]: Added.

File: parity-fix/doc/FIX.md (hunk 3 above)
Comment (jvirtanen) [inline parity-fix/doc/FIX.md]: Do you envision the FIX gateway maintaining the account information for the market participant? The trading system itself has no concept of an account.

Comment (penberg) [inline parity-fix/doc/FIX.md]: It's just auxiliary data that's passed back via ``ExecutionReport``. It's needed for FIX drop to map trades to a portfolio or an account. There's no requirement for a ClOrdID(11) to be unique across sessions so it's difficult to tell which system submitted an order without extra information provided by Account(11) .

Comment (jvirtanen) [inline parity-fix/doc/FIX.md]: Added.

File: parity-fix/doc/FIX.md (hunk 1 above)
Comment (jvirtanen) [inline parity-fix/doc/FIX.md]: Added.

**Pull request merged**